src/wms/src/main/java/org/geoserver/wms/legendgraphic/ColorMapLegendCreator.java

```diff
@@ -924,7 +934,7 @@ private BufferedImage mergeRows(Queue<BufferedImage> legendsQueue) {
         } else {
             List<RenderedImage> imgs = new ArrayList<RenderedImage>(legendsQueue);
             
-            LegendMerger.MergeOptions options = new LegendMerger.MergeOptions(imgs, (int) dx, (int) dy, (int) margin, backgroundColor, transparent, true, layout, rowWidth, 
+            LegendMerger.MergeOptions options = new LegendMerger.MergeOptions(imgs, (int) dx, (int) dy, (int) margin, 0, backgroundColor, transparent, true, layout, rowWidth, 
```

Comment:
The same as above, if I understood correctly the idea is that the label margin default value is 3, right ? But in this case the label margin value will have no effect so we set it to zero, right ?